### PR TITLE
KSQL-538 - Restructure and update docs for 4.1

### PR DESCRIPTION
**Pull request by joel-hamill**
Staging server:

http://staging-docs.confluent.io.s3-website-us-west-2.amazonaws.com/PR/587/4.1.0/ksql/docs/index.html

- Related: https://github.com/confluentinc/docs/pull/587

### CODE REVIEW

File: ksql-clickstream-demo/README.md [hunk 1]
```diff
@@ -15,11 +12,4 @@ This demo focuses on building real-time analytics of users to determine:
 
 The demo uses standard streaming functions (i.e., min, max, etc), as well as enrichment using child tables, table-stream joins and different types of windowing functionality.
```
Comment (bluemonk3y) [inline ksql-clickstream-demo/README.md]: should be 'stream-table join'

File: docs/tutorials/ksql-clickstream-demo.rst [hunk 2]
```diff
@@ -1,103 +1,99 @@
...
 
-    .. code:: bash
+Clone the KSQL GitHub repository. The demo is located in the ``ksql-clickstream-demo/`` folder.
 
-        $ git clone git@github.com:confluentinc/ksql.git
+.. code:: bash
 
-2.  Change directory to the ``ksql`` directory and compile the KSQL
-    code.
+    $ git@github.com:confluentinc/ksql.git
```
Comment (bluemonk3y) [inline docs/tutorials/ksql-clickstream-demo.rst]: clone command is missing...

File: ksql-clickstream-demo/README.md (hunk 1 above)
Comment (joel-hamill) [reply ksql-clickstream-demo/README.md]: all `.md` has been removed. changed in https://github.com/confluentinc/ksql/pull/832/files/9ad4adffabb7ac39ef0638b9ce292cca912baf4a#diff-21c12b62d71bcfca8656a765b65a581f

**Comment by joel-hamill** [comment]
@miguno i have updated the TOC per our discussion. ping for merge approval.

**Comment by joel-hamill** [comment]
@bluemonk3y can you give this another review - i have made further updates to the clickstream demo docs.

**Comment by miguno** [comment]
Question: Why is the font size of the left-hand navigation so huge?

File: docs/installation/client-server.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
+.. _install_ksql-client-server:
+
+Using KSQL in Client-Server Mode
```
Comment (miguno) [inline docs/installation/client-server.rst]: We don't want to call it different "modes".   See our offline discussion on how to structure the "Install" section.

File: docs/installation/config-ksql.rst [hunk 4]
```diff
@@ -3,77 +3,150 @@
...
+the SET statement in the CLI.
+
+To view the current configuration settings, use the ``SHOW PROPERTIES`` KSQL command.
 
 .. code:: sql
 
-    SET '<property-name>'='<property-value>';
+        ksql> SHOW PROPERTIES;
+
+Setting Default Properties
```
Comment (miguno) [inline docs/installation/config-ksql.rst]: This doesn't match our offline discussion on the Configuration section. For example, we want to clearly separate KSQL server configuration from KSQL CLI configuration.

File: docs/tutorials/ksql-basics-demo.rst [hunk 5]
```diff
@@ -0,0 +1,311 @@
+.. _ksql_quickstart:
+
+Using KSQL to Write Streaming Queries Against Kafka Data
```
Comment (miguno) [inline docs/tutorials/ksql-basics-demo.rst]: Shouldn't we also cover the Docker variant here? I forgot what we discussed offline for the v2 in this regard.

File: docs/tutorials/clickstream/index.rst [hunk 6]
```diff
@@ -1,7 +1,7 @@
-.. _ksql_clickstream:
+.. _ksql_clickstream-overview:
 
-Clickstream Analysis Demo
-=========================
+Clickstream Analysis Tutorial
```
Comment (bluemonk3y) [inline docs/tutorials/clickstream/index.rst]: shouldn't the naming of this be consistent with the docker version - which is called 'Clickstream Analysis using Docker' - so perhaps call it - Clickstream analysis?

File: docs/tutorials/clickstream/ksql-clickstream-demo.rst [hunk 7]
```diff
@@ -0,0 +1,342 @@
+.. _ksql_clickstream:
+
+Clickstream Analysis Tutorial
```
Comment (bluemonk3y) [inline docs/tutorials/clickstream/ksql-clickstream-demo.rst]: as above

File: docs/installation/config-ksql.rst (hunk 4 above)
Comment (joel-hamill) [reply docs/installation/config-ksql.rst]: Filed this to work on immediately after merging: https://confluentinc.atlassian.net/browse/DOCS-344

File: docs/installation/client-server.rst (hunk 3 above)
Comment (joel-hamill) [reply docs/installation/client-server.rst]: Filed this to work on immediately after merging: https://confluentinc.atlassian.net/browse/DOCS-344

File: docs/tutorials/ksql-basics-demo.rst (hunk 5 above)
Comment (joel-hamill) [reply docs/tutorials/ksql-basics-demo.rst]: Filed JIRA https://confluentinc.atlassian.net/browse/DOCS-345

File: docs/tutorials/clickstream/index.rst (hunk 6 above)
Comment (joel-hamill) [reply docs/tutorials/clickstream/index.rst]: Yes, i have trimmed off `tutorial`

File: docs/tutorials/clickstream/ksql-clickstream-demo.rst (hunk 7 above)
Comment (joel-hamill) [reply docs/tutorials/clickstream/ksql-clickstream-demo.rst]: @bluemonk3y yes i have trimmed off tutorial for this one, but i think it should remain above on the higher level tutorial container topic.

File: docs/tutorials/clickstream/index.rst (hunk 6 above)
Comment (joel-hamill) [reply docs/tutorials/clickstream/index.rst]: @bluemonk3y What do you think about `Clickstream Data Analysis Pipeline (Docker)` and `Clickstream Data Analysis Pipeline (Local)`?

**Comment by joel-hamill** [comment]
@miguno re https://github.com/confluentinc/ksql/pull/832#pullrequestreview-103323516, yes, this is meant to be rough TOC. i have created these JIRAs to capture the follow on fixes (as noted in the comments).

**Pull request merged**